normandy/classifier/views.py

```diff
@@ -0,0 +1,52 @@
+import hashlib
+
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+
+from normandy.base.utils import urlparams
+from normandy.recipes.models import Recipe
+
+
+class Client(object):
+    """A client attempting to fetch a set of recipes."""
+    def __init__(self, request):
+        self.request = request
+
+    @property
+    def locale(self):
+        pass
+
+    @property
+    def country(self):
+        pass
+
+    @property
+    def request_time(self):
+        return self.request.received_at
+
+
+class Bundle(object):
+    """A bundle of recipes to be sent to the client."""
+    def __init__(self, recipes=None):
+        self.recipes = recipes or []
+
+    @property
+    def ids(self):
+        return [recipe.id for recipe in self.recipes]
+
+    @property
+    def hash(self):
+        sorted_recipes = sorted(self.recipes, key=lambda r: r.filename)
```

Comment:
The recipes should be sorted when they are sorted, instead of sorting them every time.